code/modules/power/cell.dm

```diff
@@ -22,14 +22,12 @@
 	else
 		return PROCESS_KILL
 
-/obj/item/cell/update_icon()
-	cut_overlays()
+/obj/item/cell/update_overlays()
+	. = ..()
 	if(charge < 0.01)
 		return
-	else if(charge / maxcharge >= 0.995)
-		add_overlay("cell-o2")
-	else
-		add_overlay("cell-o1")
+	var/remaining = CEILING((charge / max(maxcharge, 1)) * 100, 25)
+	return "[charge_overlay]_[remaining]"
```

Comment:
. is a list that returns overlays to add
. += "Icon_state" would work but making an image or mutable appearance is better

Reply:
done.